test/e2e/framework/expect.go

```diff
@@ -51,7 +51,7 @@ type Expecter interface {
 
 // NewExpecter creates a informer-driven registry of expectations, which will
 // be triggered on every event that the informer ingests.
-func NewExpecter(informer cache.SharedIndexInformer) *expectationController {
+func NewExpecter(informer cache.SharedIndexInformer) Expecter {
```

Comment:
I'm curious why you made this change?

Reply:
I think this should return an interface but not a private class? I need to call ExpectBefore method of this interface https://github.com/kcp-dev/kcp/pull/913/files#diff-d37089ee294b6de484fc1d0d69e3ce8357b20367d45f1336df40e23a2461da7bR67

Reply:
See https://github.com/golang/go/wiki/CodeReviewComments#interfaces - it's better to return the concrete type

Reply:
gotcha, updated :)